File: android/src/main/java/com/androidphotopicker/AndroidPhotoPickerModule.java
package com.androidphotopicker;


import androidx.annotation.NonNull;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.Callback;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.module.annotations.ReactModule;


@ReactModule(name = AndroidPhotoPickerModule.NAME)
public class AndroidPhotoPickerModule extends ReactContextBaseJavaModule {
  public static final String NAME = "AndroidPhotoPicker";
  final PhotoPickerImp photoPickerImp;

  public AndroidPhotoPickerModule(ReactApplicationContext context) {
    super(context);
    photoPickerImp = new PhotoPickerImp(context);
  }


  @NonNull
  @Override
  public String getName() {
    return NAME;
  }

  @ReactMethod
  public void launchImageLibrary(final ReadableMap options, final Callback callback) {
    photoPickerImp.launchImageLibrary(options, callback);
  }
}

File: android/src/main/java/com/androidphotopicker/AndroidPhotoPickerModule.kt
package com.androidphotopicker

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.module.annotations.ReactModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.ReadableMap
import com.facebook.react.bridge.Callback

@ReactModule(name = AndroidPhotoPickerModule.NAME)
class AndroidPhotoPickerModule(reactContext: ReactApplicationContext) :
  NativeAndroidPhotoPickerSpec(reactContext) {

  override fun getName(): String {
    return NAME
  }


  @ReactMethod
  fun launchImageLibrary(options: ReadableMap, callback: Callback) {
    photoPickerImp.launchImageLibrary(options, callback)
  }

  companion object {
    const val NAME = "AndroidPhotoPicker"
  }
}
